pkg/cloudprovider/providers/openstack/openstack_client.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openstack
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+
+	"github.com/golang/glog"
+)
+
+func (os *OpenStack) NewNetworkV2() (*gophercloud.ServiceClient, error) {
+	network, err := openstack.NewNetworkV2(os.provider, gophercloud.EndpointOpts{
+		Region: os.region,
+	})
+	if err != nil || network == nil {
+		glog.Warningf("Failed to find network v2 endpoint: %v", err)
+		return nil, err
```

Comment:
This looks like it could return `nil, nil` which would be highly surprising to callers.  Luckily `openstack.NewNetworkV2` isn't written like this and always returns a non-nil network xor err.  Can we remove the `|| network == nil` clause here to make all that clearer?
(ditto below)